### PR TITLE
collect like terms with exponents

### PR DESCRIPTION
**Pull request by evykassirer**

### CODE REVIEW

File: lib/expression/step-solver/stepper.js [hunk 1]
```diff
@@ -214,6 +219,52 @@ function simplifyStepDFS(rootNodeObj) {
...
+  }
+  return false;
+}
+
+// For storing symbolic terms and performing operations on them.
+// Has a name and an exponent.
+// TODO: coefficients
+class symbolicTermObj {
+  constructor(node) {
+    if (!isSymbolicTerm(node)) {
```
Comment (evykassirer) [inline lib/expression/step-solver/stepper.js]: actually may as well just throw an error here

File: lib/expression/step-solver/stepper.js [hunk 2]
```diff
@@ -214,6 +219,52 @@ function simplifyStepDFS(rootNodeObj) {
...
+    default:
+      throw Error("Unsupported node type: " + node.type);
+  }
+  return false;
+}
+
+// For storing symbolic terms and performing operations on them.
+// Has a name and an exponent.
+// TODO: coefficients
+class symbolicTermObj {
```
Comment (aelnaiem) [inline lib/expression/step-solver/stepper.js]: Rename this to polynomial?

File: lib/expression/step-solver/stepper.js [hunk 3]
```diff
@@ -214,6 +219,52 @@ function simplifyStepDFS(rootNodeObj) {
...
+  constructor(node) {
+    if (!isSymbolicTerm(node)) {
+      this.name = '';
+      this.exp = 0;
+      return;
+    }
+    switch (node.type) {
+      case 'OperatorNode':
+        // this means it's of the same form as x^2
```
Comment (aelnaiem) [inline lib/expression/step-solver/stepper.js]: What if the node.op is not ^?

Comment (evykassirer) [inline lib/expression/step-solver/stepper.js]: Then it won't pass the `isSymbolicTerm` test at the beginning of the function.

**Pull request merged**